src/write_gcm.c

```diff
@@ -2213,6 +2289,59 @@ static monitored_resource_t *monitored_resource_create_from_fields(
                                               num_labels);
 }
 
+static monitored_resource_t *parse_monitored_resource(char *metadata,
+    int metadata_size, const char *project_id) {
+  yajl_val node;
+  char errbuf[1024];
+  node = yajl_tree_parse(metadata, errbuf, sizeof(errbuf));
```

Comment:
I wonder if we can do this with just the yajl streaming parser, without pulling in yajl_tree?..